Sources/Hyperspace/HTTP/HTTP.swift

```diff
@@ -127,53 +127,93 @@ public struct HTTP {
             self.init(data)
         }
     }
+
+    /// Represents an HTTP request
+    public struct Request: Equatable {
+
+        /// The `URL` to which the `Request` originated.
+        public let url: URL?
```

Comment:
Should this be optional? I feel like a request can definitely not have any headers or body, but it's always got a URL.

Reply:
It doesn't necessarily need to be, but because I only really have the `URLRequest` at this point - I mimicked it's structure, and `URLRequest.url` is optional.

Reply:
Makes sense.